include/commonpp/metric/reservoir/ExponentiallyDecaying.hpp

```diff
@@ -170,19 +178,15 @@ void ExponentiallyDecaying<s, a, c, m>::rescale()
     auto old_start_time = start_time_;
     start_time_ = now;
 
-    decltype(values_) values;
-    values.swap(values_);
-
     const auto time =
         std::chrono::duration_cast<ElapsedTimeUnit>(start_time_ - old_start_time)
             .count();
 
     auto scale_factor = ::exp(-ALPHA * time);
-    for (const auto& v : values)
+    for (auto& v : values_)
     {
-        auto key = v.first;
-        auto const& sample = v.second;
-        values_[key * scale_factor] = {sample.weight * scale_factor, sample.value};
+        v.mc_weight *= scale_factor;
+        v.weight *= scale_factor;
```

Comment:
does it need to be sorted again?

Reply:
no, all entries are multiplied with the same (positive) value. The order does not change.